packages/wonder-blocks-modal/src/components/drawer-launcher.tsx

```diff
@@ -0,0 +1,408 @@
+        typeof controlledOpened === "boolean"
+            ? controlledOpened
+            : uncontrolledOpened;
+
+    const saveLastElementFocused = React.useCallback(() => {
+        lastElementFocusedOutsideModalRef.current =
+            document.activeElement as HTMLElement;
+    }, []);
+
+    // Save last focused element when modal opens
+    React.useEffect(() => {
+        if (controlledOpened && !prevControlledOpened.current) {
+            saveLastElementFocused();
+        }
+        prevControlledOpened.current = controlledOpened;
+    }, [controlledOpened, saveLastElementFocused]);
+
+    const prevControlledOpened = React.useRef(controlledOpened);
+
+    const returnFocus = React.useCallback(() => {
+        // Focus on the specified element after closing the modal.
+        const focusElement = closedFocusId
+            ? document.getElementById(closedFocusId)
+            : null;
+
+        // Wait for modal to leave DOM before focusing
+        schedule.animationFrame(() => {
+            if (focusElement) {
+                focusElement.focus();
+            } else if (lastElementFocusedOutsideModalRef.current) {
+                lastElementFocusedOutsideModalRef.current.focus();
+            }
+        });
+    }, [closedFocusId, schedule]);
+
+    const handleCloseModal = React.useCallback(() => {
+        if (animated) {
+            // If component is animated, allow time for exit animations
+            setIsExiting(true);
+            setTimeout(() => {
+                setIsExiting(false);
+                if (typeof controlledOpened === "boolean") {
+                    onClose?.();
+                } else {
+                    setUncontrolledOpened(false);
+                    onClose?.();
+                }
+                returnFocus();
+            }, timingDuration);
+        } else {
+            // For non-animated case, cleanup immediately
+            if (typeof controlledOpened === "boolean") {
+                onClose?.();
+            } else {
+                setUncontrolledOpened(false);
+                onClose?.();
+            }
+            returnFocus();
+        }
+    }, [controlledOpened, onClose, returnFocus, animated, timingDuration]);
+
+    const openModal = React.useCallback(() => {
+        saveLastElementFocused();
+        setUncontrolledOpened(true);
+    }, [saveLastElementFocused]);
+
+    // Memoize drawerDialogProps to prevent unnecessary re-renders of DrawerContext consumers
+    const drawerDialogProps = React.useMemo(
+        () => ({
+            alignment,
+            animated,
+            isExiting,
+            timingDuration,
+        }),
+        [alignment, animated, isExiting, timingDuration],
+    );
+
+    const renderModal = React.useCallback(() => {
+        // Handle function-based modals
+        if (typeof modal === "function") {
+            const renderedModal = modal({
+                closeModal: handleCloseModal,
+                ...drawerDialogProps,
```

Comment:
Do the `drawerDialogProps` need to be passed in as function props still since the `DrawerDialog` component can get the props from the DialogContext now?